pkg/http/server.go

```diff
@@ -81,6 +90,43 @@ func (s *server) setupListener(opts Opts) error {
 	return nil
 }
 
+func setupUnixSocketListener(networkType string, address string) (net.Listener, error) {
+	socketOpts := strings.Split(address, ",")
+	if len(socketOpts) < 2 {
+		return net.Listen(networkType, address)
+	}
+
+	socketPath := socketOpts[0]
+	socketMode := os.FileMode(0o644)
+	for _, socketOpt := range socketOpts[1:] {
+		socketOpt := strings.SplitN(socketOpt, "=", 2)
+		if len(socketOpt) != 2 {
+			return nil, fmt.Errorf("unix socket option %s requires a value", socketOpt)
+		}
+
+		switch socketOpt[0] {
+		case "mode":
+			mode, err := strconv.ParseUint(socketOpt[1], 8, 32)
```

Comment:
imo it's better to expect base8 (ie chmod-like) rather than surprise users with a default parsing of base10, but maybe someone really prefers base10 somewhere...